app/file-upload/server/methods/sendFileMessage.js

```diff
@@ -43,15 +43,27 @@ Meteor.methods({
 			title_link_download: true,
 		};
 
+		let thumbId;
+
 		if (/^image\/.+/.test(file.type)) {
 			attachment.image_url = fileUrl;
 			attachment.image_type = file.type;
 			attachment.image_size = file.size;
 			if (file.identify && file.identify.size) {
 				attachment.image_dimensions = file.identify.size;
 			}
+
 			try {
 				attachment.image_preview = await FileUpload.resizeImagePreview(file);
+
+				const thumbBuffer = await FileUpload.createImageThumbnail(file);
+				if (thumbBuffer) {
+					const thumbnail = FileUpload.uploadImageThumbnail(file, thumbBuffer, roomId, user._id);
+					const thumbUrl = FileUpload.getPath(`${ thumbnail._id }/${ encodeURI(file.name) }`);
+					attachment.image_url = thumbUrl;
```

Comment:
we should also save thumbnail sizes on `attachment.image_dimensions` that is used to show the image on the channel, otherwise the dimensions will be wrong.

Reply:
a question to @ggazzo or @tassoevan , is there a way to use the thumbnail URL on messages list but show the full image when clicking on it? or do you think it should still not show the full image?

Reply:
Done, please review.

Reply:
@ggazzo @tassoevan ping =)

Reply:
yes

Reply:
And can we make this in a way that would make the full image be displayed if thumbnails are disabled (after the thumbnail was already generated)?

Reply:
After talking with Leti about this, she said that if an image had the thumbnail available, the better would be to show the thumbnail, always. If an image does not have a thumbnail, the full res image will be shown as today.

When clicking the image to preview, it should show the full resolution one, same as when downloading.

Since the feature managed this, probably there's not so much to update in the frontend 🤔 @sampaiodiego @pierre-lehnen-rc

Reply:
> Since the feature managed this, probably there's not so much to update in the frontend thinking @sampaiodiego @pierre-lehnen-rc

does it? I think the current implementation is showing the thumbnail when you click the image.

Reply:
Well, it manages the part of sending the FE with the appropiate data. The only thing FE would need to update is that (showing the correct image on click). I think i did a wrong reference to "this" on the last sentence 😂

Reply:
:joy: ok, so the PR is still missing some implementation right?

Reply:
Yup, it's missing some frontend implementation 😬 not sure who can take that